src/pools/xaveFxPool/fxPool.ts

```diff
@@ -181,34 +199,50 @@ export class FxPool implements PoolBase<FxPoolPairData> {
     getLimitAmountSwap(
         poolPairData: FxPoolPairData,
         swapType: SwapTypes
+    ): OldBigNumber {
+        return this._inHigherPrecision(
+            this._getLimitAmountSwap,
+            poolPairData,
+            swapType
+        );
+    }
+
+    _getLimitAmountSwap(
+        poolPairData: FxPoolPairData,
+        swapType: SwapTypes
     ): OldBigNumber {
         try {
             const parsedReserves = poolBalancesToNumeraire(poolPairData);
 
-            const alphaValue = Number(formatFixed(poolPairData.alpha, 18));
+            const alphaValue = poolPairData.alpha.div(bnum(10).pow(18));
 
-            const maxLimit = (1 + alphaValue) * parsedReserves._oGLiq * 0.5;
+            const maxLimit = alphaValue
+                .plus(1)
+                .times(parsedReserves._oGLiq)
+                .times(0.5);
```

Comment:
Hey @andreiashu 👋 
I'm reviewing the PR over here and I realized you're updating from `number` to `bignumber.js`, but still using `float points`.
Do you think it's possible to move to ethers `BigNumber` or `bigint` and update the code to use `fixed point` math?
I ask that because that's what we're going to use on our new SOR version to keep consistency.
I know this would require a bigger change, but it would help a lot to migrate things later on.
Let me know if that makes sense or if you have any questions!

Reply:
hi @brunoguerios, I appreciate you looking over this PR!

I just pushed a commit whereby we now pass the floating points as strings to `OldBigNumber`.

re moving to ethers `BigNumber` or `bigint`: you mean that, ideally, we shouldn't be using `OldBigNumber` anymore and just stick with Ethers `BigNumber`?

re `fixed point math`: can you give me an example as to how you're approaching this in the new version of SOR? We're using ABDKMath64x64.sol in our contracts, and 36 decimals seemed to be enough precision on the SOR side for us (see [_inHigherPrecision](https://github.com/balancer/balancer-sor/pull/393/files#diff-20a35a1952a13d128cda3577dbc8e0039875591f9add2bce5be5517a41e6b71aR356) and [parseFixedCurveParam](https://github.com/balancer/balancer-sor/pull/393/files?diff=unified&w=0#diff-e7e43706aa412181d4baae723ce646f76fff8b5902c9ec2d0d2b078736f61711R4) ). We have several places where we round (down) to a specific number of decimals depending on the number of decimals that is expected (see [viewRawAmount](https://github.com/balancer/balancer-sor/pull/393/files?diff=unified&w=0#diff-7c4209c57e771adf2f66e9504e10e0af132a442b927e41696ebd8effac4c1b4cR215) and [viewNumeraireAmount](https://github.com/balancer/balancer-sor/pull/393/files?diff=unified&w=0#diff-7c4209c57e771adf2f66e9504e10e0af132a442b927e41696ebd8effac4c1b4cR238)). This part was somewhat more challenging to fix in the SOR code. In our smart contracts, converting from wei to fixed point 64x64 happens several times throughout the lifecycle of a swap. Each conversion operation has its own precision loss that ultimately was compounding to higher values of loss in total in our SOR code.

Reply:
Yes, moving from `OldBigNumber` to either ethers `BigNumber` or `bigint` would be the ideal solution, because it forces you to use `fixed point` instead of `float point`.
This is how that `maxLimit` calculation would look like in fixed point math:
```
const alphaFixed = parseFixed(alphaValue.toString(), 36);
const ONE_36 = parseFixed('1', 36);
const _oGliqFixed = parseFixed(
    parsedReserves._oGLiq.toString(),
    36
);
const maxLimitFixed = alphaFixed
    .add(ONE_36)
    .mul(_oGliqFixed)
    .div(ONE_36) // I left this step explicit, but this is better done using a mulDownFixed helper function
    .div(2);
```
Of course you'd already have all variables as BigNumber and those conversions wouldn't be necessary, but I added so it's easier to follow along what each of them represents.

You can also look at how other pools are handling that within the SOR.

Gyro pool for example uses 38 decimals fixed point math, which can be found on `gyroSignedFixedPoint.ts`

Reply:
Hey @andreiashu 👋 
Thanks for sharing your progress so I can provide feedback early 😊 
Yeah, based on `calculateMicroFee`, you're definitely going in the right direction 👍
Thanks for taking the time and applying these changes!
They will help a lot to include FX pools on the new version of the SDK 😃

Reply:
hi @brunoguerios we got rid of `OldBigNumber` pretty much across our codebase. The functions that are called by the SOR (`_exactTokenInForTokenOut`, `_tokenInForExactTokenOut`, `_spotPriceAfterSwapExactTokenInForTokenOut` and the `_derivativeSpotPriceAfter[...]`) are still converting back to `OldBigNumber` as expected.

Let me know what you think